var/spack/repos/builtin/packages/flux-sched/package.py

```diff
@@ -54,16 +55,16 @@ class FluxSched(CMakePackage, AutotoolsPackage):
     version("0.18.0", sha256="a4d8a6444fdb7b857b26f47fdea57992b486c9522f4ff92d5a6f547d95b586ae")
     version("0.17.0", sha256="5acfcb757e2294a92eaa91be58ba9b42736b88b42d2937de4a78f4642b1c4933")
 
+    depends_on("c", type="build")  # generated
+    depends_on("cxx", type="build")  # generated
+
     # Avoid the infinite symlink issue
     # This workaround is documented in PR #3543
     build_directory = "spack-build"
 
     variant("docs", default=False, description="Build flux manpages and docs")
     variant("cuda", default=False, description="Build dependencies with support for CUDA")
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
-
```

Comment:
@vsoch these dependencies should be listed after variants not before according to the package style guide.

This was recently fixed in the main Spack repository, so I suspect your script may not be pulling in the updated package file before adding the new version?

Reply:
It does both, the PR here just depends on the order. Develop on our branch has the version you are referencing:

https://github.com/flux-framework/spack/blob/320c758fea1318a2e50549c32c5f1cc32ce9e371/var/spack/repos/builtin/packages/flux-sched/package.py#L64-L65

I can just change it manually here to make the change go away and that should resolve it.

Reply:
Why would the order matter? That seems like an easy thing to mess up.

Reply:
Functionally the order doesn't matter, it's a style thing that makes reviewing / reading packages easier.

Better to put all of the dependencies together when trying to understand what something depends on.

Reply:
Fixed.